accounts-db/src/io_uring/memory.rs

```diff
@@ -120,49 +131,144 @@ impl DerefMut for PageAlignedMemory {
     }
 }
 
-/// Fixed mutable view into externally allocated bytes buffer
+/// Fixed mutable view into externally allocated IO bytes buffer
+/// registered in `io_uring` for access in scheduled IO operations.
 ///
-/// It is an unsafe (no lifetime tracking) equivalent of `&mut [u8]`
-pub struct BorrowedBytesMut {
+/// It is used as an unsafe (no lifetime tracking) equivalent of `&mut [u8]`.
+pub(super) struct FixedIoBuffer {
     ptr: *mut u8,
     size: usize,
+    io_buf_index: Option<u16>,
 }
 
-impl BorrowedBytesMut {
+impl FixedIoBuffer {
     pub const fn empty() -> Self {
         Self {
             ptr: std::ptr::null_mut(),
             size: 0,
+            io_buf_index: None,
         }
     }
 
-    pub fn from_mut_slice(buf: &mut [u8]) -> Self {
-        Self {
-            ptr: buf.as_mut_ptr(),
-            size: buf.len(),
-        }
-    }
+    /// Split buffer into `chunk_size` sized `IoFixedBuffer` buffers for use as registered
+    /// buffer in io_uring operations.
+    pub unsafe fn split_buffer_chunks(
+        buffer: &mut [u8],
+        chunk_size: usize,
+    ) -> impl Iterator<Item = Self> + use<'_> {
+        assert!(
+            buffer.len() / FIXED_BUFFER_LEN <= u16::MAX as usize,
+            "buffer too large to register in io_uring"
+        );
+        let buf_start = buffer.as_ptr() as usize;
 
-    pub fn as_mut_ptr(&self) -> *mut u8 {
-        self.ptr
+        buffer.chunks_exact_mut(chunk_size).map(move |buf| {
+            let io_buf_index = (buf.as_ptr() as usize - buf_start) / FIXED_BUFFER_LEN;
+            Self {
+                ptr: buf.as_mut_ptr(),
+                size: buf.len(),
+                io_buf_index: Some(io_buf_index as u16),
+            }
+        })
     }
 
     pub fn len(&self) -> usize {
         self.size
     }
 
+    pub unsafe fn as_mut_ptr(&self) -> *mut u8 {
```

Comment:
nit: we should say here

```
/// Safety: while just returning without dereferencing a pointer is safe, this is marked unsafe so that the callers are encouraged to reason about the lifetime of the buffer.
```

Reply:
done